### DISCUSSION TITLE
Celery Not able to connect to SQS using boto3 credentials

### DISCUSSION THREAD

**Discussion by ne-api** [celery/kombu#2120] [Q&A]
I'm trying to use Celery with SQS with the below config. But I receive InvalidClientToken error. Could some one help me to resolve this?

I'm using celery[sqs]==5.3.6, Kombu==5.3.7

```
# Celery configuration
SQS_QUEUE_URL = f'https://sqs.us-east-2.amazonaws.com/{settings.aws_account}/{settings.celery_worker_queue}' 
result_backend = f'db+postgresql://{settings.pg_db_config.user}:{settings.pg_db_config.pwd}@{settings.pg_db_config.host}:5432/{settings.pg_db_config.db_name}'
pg_db_conn_string = f"postgresql://{settings.pg_db_config.user}:{settings.pg_db_config.pwd}@{settings.pg_db_config.host}:{settings.pg_db_config.port}/{settings.pg_db_config.db_name}"

session = boto3.Session(profile_name='ar-se-sbx')

app = Celery(
    'worker',
    broker_url=f"sqs://",
    backend=result_backend,
    task_default_queue=settings.celery_worker_queue,
    broker_connection_retry_on_startup=True,
    broker_transport_options={
        "region": "us-east-2",
        'predefined_queues': {
            'celery-worker-1':{
                'url': SQS_QUEUE_URL,
                'access_key_id': session.get_credentials().access_key,
                'secret_access_key': session.get_credentials().secret_key,
            }
        }
    },
    task_create_missing_queues=False,
)
```

Error:
![image](https://github.com/user-attachments/assets/2cb96bba-1b17-4f2a-9b1f-5fdc6c61a152)

**Comment by TayfurYldz** [comment]
One detail in this configuration is easy to miss: boto3.Session(profile_name=...) may be using temporary credentials, but predefined_queues only passes the access key and secret key into Kombu's SQS client in this path. A temporary credential set is not valid without its session token, so copying only those two fields can produce InvalidClientToken.

For a quick diagnosis, inspect session.get_credentials().get_frozen_credentials() and check whether token is non-null. If it is, avoid extracting that temporary credential set into predefined_queues like this. Let the normal AWS credential provider chain supply credentials to the SQS client (for example an ECS task role in production), or use Kombu's supported STS configuration when assuming a role.

There is a second thing worth checking independently: the key in predefined_queues has to be the exact Kombu queue name. The example uses celery-worker-1, while task_default_queue comes from settings.celery_worker_queue; those need to match.

So I would first reduce the configuration to the queue URL/region plus the normal AWS credential chain. If that succeeds, the failure was credential propagation rather than SQS queue configuration.
